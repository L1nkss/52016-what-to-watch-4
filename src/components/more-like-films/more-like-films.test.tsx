import * as React from "react";
import MoreLikeFilms from "./more-like-films";
import {Router} from "react-router";
import {createMemoryHistory} from "history";
import {mount} from "enzyme";

// Моки
const films = [
  {
    name: `Fantastic Beasts: The Crimes of Grindelwald`,
    backgroundImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    previewVideoLink: `img/bohemian-rhapsody.jpg`,
    src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Drama`,
    previewImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    posterImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    released: 2015,
    id: 1
  },
  {
    name: `Fantastic Beasts: The Crimes of Grindelwald`,
    backgroundImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    previewVideoLink: `img/bohemian-rhapsody.jpg`,
    src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Drama`,
    previewImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    posterImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    released: 2015,
    id: 2
  },
];

describe(`Testing More like films component`, () => {
  it(`Component should successfully rendered`, () => {
    const history = createMemoryHistory(`/sign-in`);
    const tree = mount(<Router history={history}><MoreLikeFilms films={films} /></Router>);
    expect(tree).toMatchSnapshot();
  });
});
