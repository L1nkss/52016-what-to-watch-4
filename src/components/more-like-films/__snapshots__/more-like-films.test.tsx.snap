// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing More like films component Component should successfully rendered 1`] = `
<div
  className="catalog__movies-list"
>
  <FilmCardItem
    data={
      Object {
        "backgroundImage": "img/fantastic-beasts-the-crimes-of-grindelwald.jpg",
        "genre": "Drama",
        "id": 1,
        "name": "Fantastic Beasts: The Crimes of Grindelwald",
        "posterImage": "img/fantastic-beasts-the-crimes-of-grindelwald.jpg",
        "previewImage": "img/fantastic-beasts-the-crimes-of-grindelwald.jpg",
        "previewVideoLink": "img/bohemian-rhapsody.jpg",
        "released": 2015,
        "src": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
      }
    }
  />
  <FilmCardItem
    data={
      Object {
        "backgroundImage": "img/fantastic-beasts-the-crimes-of-grindelwald.jpg",
        "genre": "Drama",
        "id": 2,
        "name": "Fantastic Beasts: The Crimes of Grindelwald",
        "posterImage": "img/fantastic-beasts-the-crimes-of-grindelwald.jpg",
        "previewImage": "img/fantastic-beasts-the-crimes-of-grindelwald.jpg",
        "previewVideoLink": "img/bohemian-rhapsody.jpg",
        "released": 2015,
        "src": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
      }
    }
  />
</div>
`;
